v2/src/features/UploadList/reducer.ts

```diff
@@ -0,0 +1,62 @@
+import { SugarFileState, Action } from './types';
+
+export default function uploadReducer(
+  state: SugarFileState[],
+  action: Action
+): SugarFileState[] {
+  switch (action.type) {
+    case 'TRY_UPLOAD': {
+      const { file, uuid } = action.payload;
+      const newFile: SugarFileState = {
+        file: file,
+        shareableLink: '',
+        uuid: uuid,
+        upload: {
+          state: true,
+          progress: 0,
+        },
+        hasError: {
+          state: false,
+        },
+      };
+      return [newFile, ...state];
+    }
+
+    case 'SET_SHAREABLE_LINK': {
+      const { uuid, shareableLink } = action.payload;
+      const newState = state.map((file) => {
+        if (file.uuid === uuid) {
+          return {
+            ...file,
+            shareableLink: shareableLink,
```

Comment:
This would trigger a re-render no?
So if it re-renders, it would re-trigger an upload?

Reply:
...unless there is a useEffect with second parameter empty `[]`.

Reply:
So these are two separated things: 
Yes it will trigger a rerender of the state changes, here I'm just updating the state of a react application

The API Triggers when I add a new item to upload, but it triggers as something different from the React Component or React State

There will be an API call everything you click the UploadButtonBase, because I'm passing the `uploadFile` onClick

The uploadfile function will do two things:
1. will dispatch TRY_UPLOAD action that will add a new item to React state
2. it will trigger the API call
   2.1.: if the API call succeeds, then Ill dispatch a new state change that will upload the shareableLink attribute of an specific item

Reply:
A very long explanation why this wouldn't cause a re-upload its here: https://github.com/sugarshare/sugarshare-extension-chrome/pull/6#discussion_r849398442